src/workflow/ArcanistPatchWorkflow.php

```diff
@@ -1247,11 +1254,33 @@ private function pullBaseTagFromStagingArea($id){
     return self::SUCCESS;
   }
 
+  // fetches hash from any remote repository has configured and also from
+  // staging if repo has one
+  private function pullHashFromAllRemotes($hash) {
+    if (!$hash) {
+      return;
+    }
+    $remotes = array();
+    list($success, $message, $staging, $staging_uri) =
+      $this->validateStagingSetup();
+    if ($success) {
+      $remotes[] = $staging_uri;
+    }
+    $repository_api = $this->getRepositoryAPI();
+    list($err, $stdin, $stderr) = $repository_api->execManualLocal('remote');
+    if (!$err) {
+      $remotes = array_merge($remotes, array_filter(explode(PHP_EOL, $stdin)));
+    }
+    foreach ($remotes as $remote) {
+      $repository_api->execManualLocal('fetch %s %s', $remote, $hash);
+    }
+  }
+
   private function mergeBranchFromStagingArea($id, $bundle) {
     list($success,
       $message, $staging, $staging_uri) = $this->validateStagingSetup();
     if (!$success) {
-      throw new ArcanistUsageException(pht($message));
```

Comment:
Why remove pht call?

Reply:
It is linting error, `pht` first parameter should be "constant"... pht `sprintf`... Doesn't make much sense also "translating" error doesn't make much sense...